src/common/NetworkPrivate.cpp

```diff
@@ -119,7 +122,10 @@ void loadUncached(const std::shared_ptr<NetworkData> &data)
                 case NetworkRequestType::Patch:
                     if (data->multiPartPayload_)
                     {
-                        assert(data->payload_.isNull());
+                        Q_ASSERT_X(
```

Comment:
The same message can be achieved with normal `assert` and doesn't add extra complexity (e.g. asserts firing because qt version was built with force-asserts, or the message not coming through because it uses qlogging)

e.g. `assert((data->payload_.isNull()) && "bla bla");`

Depending on who the assert is for, a comment would also be equally useful

Reply:
My thinking boils down more to: Let's not replace 3 of 152 asserts when there's no need for the feature you're looking for.
Adding the extra information could just as well be done through a comment or through our standard `assert`